pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -592,19 +659,39 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 		return actionContinue{provResult.RequeueAfter}
 	}
 
+	pendingRebootSince := info.host.Status.Provisioning.PendingRebootSince
+	lastPoweredOn := info.host.Status.Provisioning.LastPoweredOn
+
+	desiredPowerOnState := info.host.Spec.Online
+	isInRebootProcess := lastPoweredOn.Before(pendingRebootSince) || (!pendingRebootSince.IsZero() && lastPoweredOn.IsZero())
+
+	if isInRebootProcess {
+		suffixlessAnnotationExists, shouldHoldPowerOff := getRebootAnnotations(info.host)
+
+		if suffixlessAnnotationExists && !info.host.Status.PoweredOn {
+			delete(info.host.Annotations, rebootAnnotationPrefix)
+			return actionContinue{}
```

Comment:
This isn't going to do what you expect. The Reconcile() method only saves the status fields, which do not include metadata like the annotations. So you do need to call Update() before returning.

Reply:
roger, fixed.